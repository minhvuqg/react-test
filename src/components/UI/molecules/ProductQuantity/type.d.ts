export interface IProps {
  quantity: number
  incrementQuantity?: () => void
  decrementQuantity?: () => void
}
