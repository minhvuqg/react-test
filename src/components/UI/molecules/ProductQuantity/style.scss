.quantity-actions {
  display: flex;
  justify-content: space-around;
  width: 100%;
  align-items: center;
  max-width: 100px;

  button {
    border-radius: 50%;
    border: 1px solid;
    cursor: pointer;
    width: 24px;
    height: 24px;
  }
}
