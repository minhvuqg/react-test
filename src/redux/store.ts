import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
