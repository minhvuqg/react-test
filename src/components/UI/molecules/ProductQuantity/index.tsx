import { FC } from 'react'

import { IProps } from './type.d'

import './style.scss'

const ProductQuantity: FC<IProps> = ({
  quantity,
  incrementQuantity,
  decrementQuantity
}: IProps): JSX.Element => {
  return (
    <div className="quantity-actions">
      <button data-testid="decrementBtn" onClick={decrementQuantity}>
        -
      </button>
      <span data-testid="quantity">{quantity}</span>
      <button data-testid="incrementBtn" onClick={incrementQuantity}>
        +
      </button>
    </div>
  )
}

export { ProductQuantity }
