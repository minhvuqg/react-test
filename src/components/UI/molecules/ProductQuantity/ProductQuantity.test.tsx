import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ProductQuantity } from '../ProductQuantity'
import store from 'redux/store'

it('render detail correctly', () => {
  const detailEl = TestRenderer.create(
    < Provider store={store} >
      <ProductQuantity quantity={3}
        incrementQuantity={() => { }}
        decrementQuantity={() => { }} />
    </Provider >).toJSON()
  expect(detailEl).toMatchSnapshot()
})
