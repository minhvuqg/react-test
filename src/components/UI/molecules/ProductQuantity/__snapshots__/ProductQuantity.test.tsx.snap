// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail correctly 1`] = `
<div
  className="quantity-actions"
>
  <button
    data-testid="decrementBtn"
    onClick={[Function]}
  >
    -
  </button>
  <span
    data-testid="quantity"
  >
    3
  </span>
  <button
    data-testid="incrementBtn"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
